datapath: Remove 2005 route table for IPv6

jschwinger233 opened the pull request:
This reverts 3ed62d51d5ac27bbe2c60372850d4ec78c17bb6e for IPv6 part only, as issue #21954 has been resolved by #24208.

Another PR #24807  removes 2005 route table for IPv4.

Signed-off-by: Zhichuan Liang <gray.isovalent.com>

```release-note
Fix broken IPv6 connectivity from outside to NodePort service when L7 ingress policy applied by removing PROXY_RT route table.
```

jschwinger233 committed:
datapath: remove 2005 route table for ipv6 only

This reverts 3ed62d5 partially and only removes ipv6 2005 route table.

Signed-off-by: Zhichuan Liang <gray.liang@isovalent.com>

christarazi commented on test/k8s/services.go:
```diff
@@ -416,6 +416,27 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`,
 			}
 		})
 
+		Context("with L7 policy", func() {
```
Is it possible to test within the Cilium CLI connectivity test or do we need to add more tests to Ginkgo here?

jschwinger233 replied:
Commit dropped, because I just realized this e2e test is already there at https://github.com/cilium/cilium/blob/ca5a9d04b55b6dbb08d6b8d3c1b4a96f78f7df84/test/k8s/services.go#L108-L119

pchaigno reviewed (approved):
LGTM.

Out of curiosity, why did you separate the PRs for IPv4 and IPv6?

> This PR removes 2005 route table for IPv6, and changes datapath for return IPv6 traffic from L7 proxy.

Our release notes usually don't have a subject; it is understood to be "this release". Also nit: the comma before "and" is superfluous since both clauses share the same subject.

jschwinger233 committed:
test: enable ipv6 test for reaching nodeport from outside

Those test cases were temporarily deleted by #24807 to pass CI, and this
commit adds them back.

Signed-off-by: Zhichuan Liang <gray.liang@isovalent.com>
